olive/auto_optimizer/__init__.py

```diff
@@ -0,0 +1,126 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+import logging
+from copy import deepcopy
+from enum import Enum
+from typing import Dict, List, Optional
+
+from olive.auto_optimizer.regulate_mixins import RegulatePassConfigMixin
+from olive.common.config_utils import ConfigBase
+from olive.common.pydantic_v1 import validator
+from olive.data.config import DataConfig
+from olive.evaluator.olive_evaluator import OliveEvaluatorConfig
+from olive.hardware.accelerator import AcceleratorSpec
+from olive.model import ModelConfig
+
+logger = logging.getLogger(__name__)
+
+
+class Precision(str, Enum):
+    FP32 = "fp32"
+    FP16 = "fp16"
+    INT8 = "int8"
+    INT4 = "int4"
+
+
+class AutoOptimizerConfig(ConfigBase):
+    # opt_level
+    # 0: look up template mapping
+    # TODO(anyone): support more opt_level
+    opt_level: int = 0
+
+    # used to disable auto optimizer when user only want to evaluate input model
```

Comment:
If user only wants to evaluate input model, will they add `auto_optimizer_config` in configuration?

Reply:
Yes, they need to disable auto optimizer with `disable_auto_optimizer` as True.

Reply:
I expect if they do not add `auto_optimizer_config` to config file, auto optimizer will be automatically disabled?

Reply:
I suppose that should not be the common use case. Let us think this, for most of the users who leverage olive to optimize model, it would be simpler user just provide model, then the auto optimizer will begin to search.

Reply:
Updated with logics of: 
> If pass list is provided by the user then run user provided pass list by default. Ignore user provided pass list only if auto opt config is also provided and auto opt is not disabled explicitly.

> if pass list is not provided and run auto opt unless auto opt is explicitly disabled in the auto opt config.

> If user only wants to evaluate input model then he should not provide pass list and ( disable auto opt or not provide auto opt config)